sklearn/model_selection/_validation.py

```diff
@@ -573,13 +573,33 @@ def _fit_and_score(estimator, X, y, scorer, train, test, verbose,
             raise ValueError("error_score must be the string 'raise' or a"
                              " numeric value. (Hint: if using 'raise', please"
                              " make sure that it has been spelled correctly.)")
-
     else:
-        fit_time = time.time() - start_time
-        test_scores = _score(estimator, X_test, y_test, scorer)
-        score_time = time.time() - start_time - fit_time
-        if return_train_score:
-            train_scores = _score(estimator, X_train, y_train, scorer)
+        try:
+            fit_time = time.time() - start_time
+            test_scores = _score(estimator, X_test, y_test, scorer,
+                                 error_score)
+            score_time = time.time() - start_time - fit_time
+            if return_train_score:
+                train_scores = _score(estimator, X_train, y_train, scorer,
+                                      error_score)
+        except Exception:
+            if error_score == 'raise':
+                raise
+            elif isinstance(error_score, numbers.Number):
+                score_time = time.time() - start_time - fit_time
+                test_scores = {name: error_score for name in scorer}
+                if return_train_score:
+                    train_scores = test_scores.copy()
+                warnings.warn("Scoring failed. The score on this train-test "
+                              "partition for these parameters will be set "
+                              "to %f. Details: \n%s" %
+                              (error_score, format_exc()),
+                              UserWarning)
+            else:
+                raise ValueError("error_score must be the string 'raise' or a "
+                                 "numeric value. (Hint: if using 'raise', "
+                                 "please make sure that it has been "
+                                 "spelled correctly.)")
```

Comment:
we can't check this in the beginning of _fit_and_score right? Because we should be checking error_score only when fit or score failed.. we cannot check it at the beginning when we don't know if the fit/score actually fails or passes.

Reply:
Validating the input `error_score` in the beginning is not too costly and allows the rest of the code to not constantly need to check:

```py
if not (isinstance(error_score, numbers.Number) or error_score == 'raise'):
	raise ValueError("error_score must be a string 'raise' or ...")

progress_msg = ""
...
```

Reply:
Lets say error_score is 'raise', but fit and score function doesn't fail. Then we incorrectly raise the ValueError if we add it to the beginning of the function and check it everytime. We should check only when fit/score fails

Reply:
The thing is: if fit and score are passing, then irrespective of error_score, we have to return the scores received. But if we check error_score at the beginning, we deviate from that behavior.

Reply:
> Lets say error_score is 'raise', but fit and score function doesn't fail.

If `error_score` is `'raise`' then the above condition will not raise. It only raise if `error_score` is not a number and its not equal to `'raise'`.

> The thing is: if fit and score are passing, then irrespective of error_score, we have to return the scores received. But if we check error_score at the beginning, we deviate from that behavior.

In our public api documentation for [cross_validate_score](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_val_score.html#sklearn.model_selection.cross_val_score) or [GridSearchCV](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html#sklearn.model_selection.GridSearchCV) it states that `error_score` must be numeric or `'raise'`. If a user passes an invalid `error_score` I think we should have already raised an error without doing any calculation.

Reply:
> The thing is: if fit and score are passing, then irrespective of error_score, we have to return the scores received. But if we check error_score at the beginning, we deviate from that behavior.

This is a valid concern. For this specific case, I think doing input validation first will be better long run. Let's see a user provides `error_score='wow'`, I would want that to fail right away. Otherwise a user would think that `error_score='wow'` is a valid input which is not what the API docs states.

Reply:
Updated. Please review